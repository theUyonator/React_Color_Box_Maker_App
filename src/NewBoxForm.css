.NewBoxForm {
    margin: 50px;
    padding:10px
}

.NewBoxForm-label {
    margin: 10px
}

.NewBoxForm-input {
    margin: 15px
}

.NewBoxForm-btn {
    margin: 10px;
    cursor: pointer;
    color: white;
    background-color: green
}

