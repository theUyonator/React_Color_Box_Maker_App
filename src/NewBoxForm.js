import React, { useState } from "react";
import "./NewBoxForm.css";

/** Form for creating a new box to add to the box list.
 *
 * Has state for the width, height and background color  of the box; on submission,
 * sends {width, height, backgroundColro} to fn rec'd from parent.
 *
 */

const NewBoxForm = ({ addBox }) => {
  const INITIAL_STATE = { width: 0, height: 0, backgroundColor: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  /** Send {width, height, backgroundColor} to parent
   *    & clear form. */

  const handleSubmit = evt => {
    evt.preventDefault();
    addBox(formData);
    setFormData(INITIAL_STATE);
  };

  /** Update local state w/curr state of input elem */

  const handleChange = evt => {
    const { name, value }= evt.target;
    setFormData(fData => ({
      ...fData,
      [name]: value
    }));
  };

  /** render form */

  return (
    <form className="NewBoxForm" onSubmit={handleSubmit}>
      <label className="NewBoxForm-label" htmlFor="width">Width:</label>
      <input
        type="number"
        id="width"
        name="width"
        value={formData.width}
        className="NewBoxForm-input"
        onChange={handleChange}
      />

      <label className="NewBoxForm-label" htmlFor="height">Height:</label>
      <input
        type="number"
        id="height"
        name="height"
        value={formData.height}
        className="NewBoxForm-input"
        onChange={handleChange}
      />

    <label className="NewBoxForm-label" htmlFor="backgroundColor">Background Color:</label>
      <input
        type="text"
        id="backgroundColor"
        name="backgroundColor"
        value={formData.backgroundColor}
        className="NewBoxForm-input"
        onChange={handleChange}
      />

      <button id="newBoxBtn" className="NewBoxForm-btn">Add a new box!</button>
    </form>
  );
};

export default NewBoxForm;