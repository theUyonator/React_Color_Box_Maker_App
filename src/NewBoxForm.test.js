import React from "react";
import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";


describe("<NewBoxForm /> rendering", () => {

    it("renders correctly without crashing", () => {
        render(<NewBoxForm />)
    });
    
    it("matches snapshot", () => {
        const { asFragment } = render(<NewBoxForm />);
        expect(asFragment()).toMatchSnapshot();
    });
    
    });