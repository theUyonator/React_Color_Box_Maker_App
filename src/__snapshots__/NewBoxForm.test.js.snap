// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewBoxForm /> rendering matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      class="NewBoxForm-label"
      for="width"
    >
      Width:
    </label>
    <input
      class="NewBoxForm-input"
      id="width"
      name="width"
      type="number"
      value="0"
    />
    <label
      class="NewBoxForm-label"
      for="height"
    >
      Height:
    </label>
    <input
      class="NewBoxForm-input"
      id="height"
      name="height"
      type="number"
      value="0"
    />
    <label
      class="NewBoxForm-label"
      for="backgroundColor"
    >
      Background Color:
    </label>
    <input
      class="NewBoxForm-input"
      id="backgroundColor"
      name="backgroundColor"
      type="text"
      value=""
    />
    <button
      class="NewBoxForm-btn"
      id="newBoxBtn"
    >
      Add a new box!
    </button>
  </form>
</DocumentFragment>
`;
